### ISSUE TITLE
feat: 장기 스윙 전략(long_term_swing) 신규 추가 — 진득한 매매로 수수료 적자 구조 해소

### ISSUE THREAD

**Issue by SeungMinK** [SeungMinK/cryptobot#226]
## 한 달 운영 결과 정직한 진단

**기간**: 2026-04-04 ~ 2026-05-06 (33일)  
**자본**: 50만원 입금 → 47만원 (**−6%, −3만원 손실**)  
**매매**: 매도 **265건** (일평균 8.0건)

| 지표 | 값 | 평가 |
|---|---|---|
| 승률 | 66.8% (96승/47패는 4/1~4/20 구간 통계) | ✅ 좋음 |
| 평균 수익 | +1.31% | — |
| 평균 손실 | −2.46% | — |
| **손익비** | **0.53** | ❌ 1.5 목표의 1/3 |
| **건당 EV** | **+0.058%** | ❌ |
| **왕복 수수료** | **0.1%** | — |

### 핵심 진단

**건당 기대수익(+0.058%) < 왕복 수수료(0.1%)** → **구조적 적자**.  
승률은 67%로 좋지만 매매할 때마다 평균 0.04% 손해. 1000번 매매 = 4% 손실 보장.

큰 손실 4건(−7% 이하)이 -44,740원 — 한 달 손해의 1.5배. **NEWT 한 종목만 -31,056원**. 알트 잡쓰레기 매매 + 잦은 매매 + 큰 손실 1~2건 = 적자 누적.

## 사용자 인사이트 (이 PR의 출발점)

> 한달 돌리면서 보니깐, 너무 잦은 매도랑 매수를 하는것보다 쫌 진득하게 기다리는쪽으로하는게 좋아보이는데  
> 결과론적으로 4월 초에 비트코인이 1억3백만원이였는데 그때 저점이라고 판단하고 매수해놓고 지금까지 들고있었으면 지금 1억2천이야

**스캘핑(분~시간 보유, 일 8건) → 스윙 트레이딩(일~주 보유, 월 1~3건)** 으로 전환 검토.

### 진입 조건 (3중 필터, 모두 충족)
- 공포탐욕지수(F&G) ≤ **30** (극도 공포)
- 가격 < 50일 MA (저평가 영역)
- RSI(14d) ≤ 45 (약한 과매도)

### 청산 조건 (하나만 충족)
- F&G ≥ 70 (시장 과열, 차익실현)
- ROI ≥ +20%
- 보유 14일+ AND 가격 < MA20 (추세 이탈)
- stop_loss = -15% (BaseStrategy 공통, 진득 보유라 더 넓게)

### 레퍼런스
- **공포탐욕지수**: alternative.me/fng (무료 공공 API, 5년+ 데이터)
- **Buy the Fear, Sell the Greed**: Nasdaq 백테스트 +1,145% (vs Buy&Hold +1,046%)
- **DCA + 공포 마켓**: 12년 백테스트 +6,712%
- **Golden Cross / 200일 MA**: 학술 표준 추세 판단

## 백테스트 결과 (122일, 2025-12 ~ 2026-04 약세장)

| 코인 | bb_rsi (현재) | **long_term_swing (신규)** | Buy&Hold |
|---|---|---|---|
| BTC | **+3.42%** (2건) | -11.85% (4건) | -22.61% |
| ETH | **+3.51%** (2건) | -49.62% (4건) | -23.98% |
| XRP | **+15.43%** (2건) | -2.84% (5건) | -27.65% |
| **평균** | **+7.45%** | -21.44% | -24.75% |

### ⚠️ 충격적 발견

**bb_rsi가 메이저 3종 일봉 백테스트에선 압승**. swing보다 좋고 Buy&Hold도 이김.

→ **실측 -21,043원 적자의 진짜 원인**: bb_rsi 자체가 아니라 **알트(NEWT 등) 마구잡이 매매**.

### swing의 한계
- ETH에서 갭다운 stop -15% 두 번 → -49.62% 큰 손실
- 122일 약세장 한 번뿐 (통계 부족)
- 일봉 기준이라 분봉 환경 검증 필요

## 결정: 옵션으로 추가, **활성화 보류**

코드 추가하지만 `is_active=False`. 기존 bb_rsi 유지. 사용자가 직접 결정 후 활성화.

```bash
# 활성화 (사용자 명령)
.venv/bin/python scripts/switch_strategy.py --to long_term_swing
bash scripts/stop_daemon.sh && bash scripts/start_daemon.sh

# 롤백
.venv/bin/python scripts/switch_strategy.py --to bb_rsi_combined
```

## 변경 파일

| 파일 | 변경 |
|---|---|
| `src/cryptobot/strategies/long_term_swing.py` | **신규** 전략 클래스 |
| `src/cryptobot/strategies/registry.py` (간접) | 자동 등록 |
| `src/cryptobot/bot/strategy_selector.py` | STRATEGY_CLASSES 등록 |
| `src/cryptobot/data/database.py` | seed 데이터 추가 (`is_active=False`) |
| `src/cryptobot/backtest/reporter.py` | SWEEP_CONFIGS 추가 |
| `scripts/switch_strategy.py` | **신규** 활성 전략 전환 도구 |
| `tests/test_long_term_swing_226.py` | **신규** 12건 단위 테스트 |
| 기존 테스트 3건 | 전략 개수 10 → 11 갱신 |

## 다음 PR 후보 (사용자 결정에 따라)

1. **메이저 코인 화이트리스트 강화** ← 진짜 레버 (실측 적자 직접 해결)
   - bb_rsi + BTC/ETH/XRP/SOL만 매매
   - 알트 코인 자동 제외 → NEWT 같은 큰 손실 차단
   
2. 분봉 ATR/ADX 재실험 (#216 분봉 데이터 쌓이면)
3. 뉴스 + 공포탐욕 통합 진입 필터 (사용자 아이디어)
4. swing 전략 BTC/XRP 전용 + paper trading 검증

## 학습 노트 (블로그 레퍼런스)

- 매매 빈도 ≠ 수익. 수수료 0.1% 못 넘는 매매는 통계적 적자
- 승률 67%는 보기엔 좋지만 손익비 0.53이면 장기 마이너스 (Kelly Criterion 기준 매매할수록 손해)
- 코인 선별이 전략 선택보다 더 큰 영향. NEWT 한 종목 = 한 달 손해의 1.5배
- 일봉 백테스트는 갭다운/슬리피지 무시. 분봉 검증 필수
- 공포탐욕지수는 무료 공공 데이터 (alternative.me)
- 헷지펀드 평균도 BTC 단순 보유 못 이김 (2024 VisionTrack +53%, BTC +120%)

**Comment by SeungMinK** [comment]
PR #227 머지 — 비활성으로 보존
